src/Http/Http/src/Features/FormFeature.cs

```diff
@@ -101,10 +101,10 @@ public IFormCollection ReadForm()
 
         if (!HasFormContentType)
         {
-            throw new InvalidOperationException("Incorrect Content-Type: " + _request.ContentType);
+            throw new InvalidOperationException("Form Collection properties can only be accessed from a POST request");
```

Comment:
The error message was correct before, we only checked the content-type, not the method. This error might make sense specifically if the Content-Type is missing. E.g. "This request does not have a Content-Type header. Forms are available from requests with bodies like POSTs and a form Content-Type of either `application/x-www-form-urlencoded` or `multipart/form-data`"